packages/cli-kit/src/common/result.ts

```diff
@@ -0,0 +1,37 @@
+export type Result<T, TError = Error> = {ok: true; value: T} | {ok: false; error: TError}
+
+export type ResultAsync<T, TError = Error> = Promise<Result<T, TError>>
+
+export const ok = <T, TError = Error>(value: T): Result<T, TError> => {
```

Comment:
No strong preference here, but have you considered naming these `Ok` and `Err` instead of `ok` and `err`?

Reply:
But these are methods, does naming convention in Typescript recommends recommend capitalizing?